Source/Mothership/TMoth.cpp

```diff
@@ -938,6 +1021,246 @@ void TMoth::StopTwig()
 
 //------------------------------------------------------------------------------
 
+
+
+//------------------------------------------------------------------------------
+// Instrumentation Methods
+//------------------------------------------------------------------------------
+
+// Instrumentation initialisation
+//void TMoth::InstrumentationInit(void)
+//{
+//    
+//}
+
+// Gracefully tear down the Instrumentation map
+void TMoth::InstrumentationEnd(void)
+{
+    WALKMAP(uint32_t,TM_Instrumentation*,InstrMap,I)
+        delete I->second;
+}
+
+
+// Handle an instrumentation message
+unsigned TMoth::InstrumentationHandler(P_Msg_t* msg)
+{
+    TM_Instrumentation* instr;
+    uint32_t srcAddr = msg->header.pinAddr;
+    
+    // Pointer to the Instrumentation
+    P_Instr_Msg_Pyld_t* instrMsg = reinterpret_cast<P_Instr_Msg_Pyld_t*>(msg->payload);
+    
+    std::ofstream tFile;        // Thread instrumentation file
+    // Set Filename
+    std::ostringstream fName;
+    fName << "instrumentation_thread_" << srcAddr << ".csv";

```

Comment:
Thoughts on standardising this name? It's not in the documentation. I also recommend adding the name of the application, the hardware address of the thread (hex), and the datetime. Would be even better to put the outputs in a separate directory to avoid filling up `bin`, at least for now. I recommend `~/.orchestrator/instrumentation/`.  :sweet_potato:

Reply:
It's not in the docs because the docs are softswitch whereas the generation of this file is Mothership. The Mothership-side implementations really is much earlier/more prototype than the rest of the implementation.

This is partly due to the pending re-write of the Mothership, and partly because it was a case of "just get it working" after de-crufting the mothership.

Reply:
As discussed with @mvousden, I will change the output directory to `~/.orchestrator/instrumentation/` and clear contents on task run/TMoth start

Reply:
Implemented in b5664a1, I think that is the last of it and ready for a re-review?

Reply:
Implemented in b5664a1, I think that is the last of it and ready for a re-review?